test/e2e/external_gateways.go

```diff
@@ -1008,17 +1134,57 @@ EOF
 				framework.ExpectNoError(err, "failed to setup FRR peer %s in %s", a, containerName)
 			}
 		}
-		cmd := []string{"docker", "exec", containerName, "/usr/lib/frr/frrinit.sh", "start"}
+		cmd := []string{containerRuntime, "exec", containerName, "/usr/lib/frr/frrinit.sh", "start"}
 		_, err := runCommand(cmd...)
 		framework.ExpectNoError(err, "failed to start frr in %s", containerName)
 	}
 }
 
 func isBFDPaired(container, peer string) bool {
-	res, err := runCommand("docker", "exec", container, "bash", "-c", fmt.Sprintf("vtysh -c \"show bfd peer %s\"", peer))
+	res, err := runCommand(containerRuntime, "exec", container, "bash", "-c", fmt.Sprintf("vtysh -c \"show bfd peer %s\"", peer))
 	framework.ExpectNoError(err, "failed to check bfd status in %s", container)
 	if strings.Contains(res, "Status: up") {
 		return true
 	}
 	return false
 }
+
+// When running on host network we clean the routes and ips we added previously
+func cleanRoutesAndIPs(containerName string, addressesv4, addressesv6 gatewayTestIPs) {
+	testIPv6 := false
+	testIPv4 := false
+
+	if addressesv6.srcPodIP != "" && addressesv6.nodeIP != "" {
+		testIPv6 = true
+	}
+	if addressesv4.srcPodIP != "" && addressesv4.nodeIP != "" {
+		testIPv4 = true
+	}
+
+	if !testIPv4 && !testIPv6 {
+		return
+	}
+
+	var err error
+	if testIPv4 {
+		ginkgo.By(fmt.Sprintf("Removing the destination ips from %s", containerName))
+		for _, address := range addressesv4.targetIPs {
+			_, err = runCommand(containerRuntime, "exec", containerName, "ip", "address", "del", address+"/32", "dev", "lo")
+			framework.ExpectNoError(err, "failed to del the loopback ip to dev lo on the test container %s", containerName)
+		}
+
+		ginkgo.By(fmt.Sprintf("Removing the route from %s to the src pod", containerName))
+		_, err = runCommand(containerRuntime, "exec", containerName, "ip", "route", "del", addressesv4.srcPodIP, "via", addressesv4.nodeIP)
+		framework.ExpectNoError(err, "failed to del the pod host route on the test container %s", containerName)
+	}
+	if testIPv6 {
+		ginkgo.By(fmt.Sprintf("Removing the the destination ips to %s (ipv6)", containerName))
+		for _, address := range addressesv6.targetIPs {
```

Comment:
Something like this IMO is cleaner as it avoids duplication:
~~~
// When running on host network we clean the routes and ips we added previously
func cleanRoutesAndIPs(containerName string, addressesv4, addressesv6 gatewayTestIPs) {
        testIPv6 := addressesv6.srcPodIP != "" && addressesv6.nodeIP != ""
        testIPv4 := addressesv4.srcPodIP != "" && addressesv4.nodeIP != ""

        if !testIPv4 && !testIPv6 {
                return
        }

        if testIPv4 {
                cleanRoutesAndIPsForFamily(containerName, 4, addressesv4)
        }
        if testIPv6 {
                cleanRoutesAndIPsForFamily(containerName, 6, addressesv6)
        }    
}

func cleanRoutesAndIPsForFamily(containerName string, family int, addresses gatewayTestIPs) {
        var err error
        addressSelector := fmt.Sprintf("-%d", family)
        mask := 32
        if family == 6 {
                mask = 128
        }    
        ginkgo.By(fmt.Sprintf("Removing the destination ips from %s (IPv%d)", containerName, family))
        for _, address := range addresses.targetIPs {
                _, err = runCommand(containerRuntime, "exec", containerName, "ip", addressSelector, "address", "del", address+"/"+mask, "dev", "lo")
                framework.ExpectNoError(err, "failed to del the loopback ip to dev lo on the test container %s", containerName)
        }    

        ginkgo.By(fmt.Sprintf("Removing the route from %s to the src pod (IPv%d)", containerName, family))
        _, err = runCommand(containerRuntime, "exec", containerName, "ip", addressSelector, "route", "del", addresses.srcPodIP, "via", addresses.nodeIP)
        framework.ExpectNoError(err, "failed to del the pod host route on the test container %s", containerName)
}
~~~

Reply:
thanks, used this